operator/src/main/java/com/github/streamshub/console/ConsoleReconciler.java

```diff
@@ -0,0 +1,170 @@
+        @Dependent(
+                name = ConsoleSecret.NAME,
+                type = ConsoleSecret.class),
+        @Dependent(
+                name = ConsoleService.NAME,
+                type = ConsoleService.class),
+        @Dependent(
+                name = ConsoleIngress.NAME,
+                type = ConsoleIngress.class,
+                dependsOn = {
+                        ConsoleService.NAME
+                },
+                readyPostcondition = IngressReadyCondition.class),
+        @Dependent(
+                name = ConsoleDeployment.NAME,
+                type = ConsoleDeployment.class,
+                dependsOn = {
+                        ConsoleClusterRoleBinding.NAME,
+                        ConsoleSecret.NAME,
+                        ConsoleIngress.NAME,
+                        PrometheusService.NAME
+                },
+                readyPostcondition = DeploymentReadyCondition.class),
+})
+@CSVMetadata(
+        provider = @Provider(name = "StreamsHub", url = "https://github.com/streamshub"),
+        description = "StreamsHub console provides a user interface for managing and monitoring your streaming resources")
+public class ConsoleReconciler implements EventSourceInitializer<Console>, Reconciler<Console>, Cleaner<Console> {
+
+    @Override
+    public Map<String, EventSource> prepareEventSources(EventSourceContext<Console> context) {
+        return Collections.emptyMap();
+    }
+
+    @Override
+    public UpdateControl<Console> reconcile(Console resource, Context<Console> context) {
+        var result = context.managedDependentResourceContext().getWorkflowReconcileResult();
+        var status = resource.getOrCreateStatus();
+
+        var readyCondition = Optional.ofNullable(status.getConditions())
+                .flatMap(conditions -> conditions.stream()
+                        .filter(c -> "Ready".equals(c.getType()))
+                        .findFirst())
+                .orElseGet(() -> {
+                    var condition = new ConditionBuilder()
+                            .withType("Ready")
+                            .withLastTransitionTime(Instant.now().toString())
+                            .build();
+                    status.getConditions().add(condition);
+                    return condition;
+                });
+
+        var notReady = result.map(r -> r.getNotReadyDependents());
+        boolean isReady = notReady.filter(Collection::isEmpty).map(r -> Boolean.TRUE)
+                .orElse(Boolean.FALSE);
+
+        String readyStatus = isReady ? "True" : "False";
+
+        if (!readyStatus.equals(readyCondition.getStatus())) {
+            readyCondition.setStatus(readyStatus);
+            readyCondition.setLastTransitionTime(Instant.now().toString());
+        }
+
+        if (isReady) {
+            readyCondition.setReason(null);
+            readyCondition.setMessage("All resources ready");
+        } else {
+            readyCondition.setReason("DependentsNotReady");
```

Comment:
I've just learnt another UK/US spelling variation :)

Reply:
Are you referring to dependent vs dependant? IMO this isn't the right spelling even for US in my experience.

Reply:
But I just copied it to match the method `getNotReadyDependents`